### PR TITLE
Anshu-Assignment-33-1hr-Done

### PR DESCRIPTION
**Pull request by AnshuKJun20**
image

### CODE REVIEW

File: src/anshuKJun20/ExtractItemIdFromString.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+				}
+			} catch (Exception e) {
+			}
+		}
+		System.out.println(" Item id is " + num);**/
+		
+		// second Approach
+		String sum = ""; int count = 0;
+		if(input.contains("itemid=")) {
+		int index = input.indexOf("12345");
```
Comment (PranitaPJun20) [inline src/anshuKJun20/ExtractItemIdFromString.java]: You can find substring like lastIndexOf("itemId=").Now try to find item id value without using character class method.
Either you can use replace ("itemId=", "")  or split method.

**Review by PranitaPJun20 — COMMENTED** [review]
Please work on comments

File: src/anshuKJun20/ExtractItemIdFromString.java (hunk 1 above)
Comment (AnshuKS20) [reply src/anshuKJun20/ExtractItemIdFromString.java]: Changes has been done.

File: src/anshuKJun20/ExtractItemIdFromString.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			for (int index = 0; index < array.length; index++) {
+				try {
+					number += Integer.parseInt(array[index]);
+				} catch (NumberFormatException ne) {
+
+				}
+			}
+			System.out.println("Item id in given string is: " + number);
+		} else {
+			System.out.println("Itemid is 0");
```
Comment (PranitaPJun20) [inline src/anshuKJun20/ExtractItemIdFromString.java:43]: Instead of this you can write "Item ID is not available"

**Review by PranitaPJun20 — APPROVED** [review]
Working as expected.

File: src/anshuKJun20/ExtractItemIdFromString.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		 **/
+
+		// second Approach
+		String number = "";
+		int count = 0;
+		if (input.contains("itemid=")) {
+			String substr = input.substring(input.lastIndexOf("itemid="), input.lastIndexOf("/"));
+			if (substr.contains("empid")) {
+				substr = substr.substring(0, substr.lastIndexOf("&"));
+			}
```
Comment (KrishnaKTechnocredits) [inline src/anshuKJun20/ExtractItemIdFromString.java:32]: if (input.contains("itemId=")) {
			String str = input.split("itemId=")[1];
			String output = "";
			for (int index = 0; index < str.length(); index++) {
				if (Character.isDigit(str.charAt(index))) {
					output += str.charAt(index);
				} else {
					break;   // with this code you don't need to check for empID. 
				}
			}
			System.out.println("Item id is " + output);
		} else {
			System.out.println("No itemID present.");
		}

This code is more optimized.

**Pull request merged**